### ISSUE TITLE
"Provider configuration not present" with azurerm when using a provider alias

### ISSUE THREAD

**Issue by marco-svitol** [microsoft/azure-pipelines-terraform#96]
With TerraformTaskVx@x if I use an alias provider such as:

```
provider "azurerm" {
  subscription_id = var.subscriptionId
  tenant_id = var.tenantId
  features {}
}

provider "azurerm" {
  alias = "sharedsubscription"
  subscription_id = var.sharedSubscriptionId
  tenant_id = var.sharedTenantId
  features {}
}

resource "azurerm_virtual_network_peering" "spoke2nonprod_peering" {
  provider = azurerm.sharedsubscription
  name                      = "spoke2nonprod_peering"
  resource_group_name       = data.azurerm_virtual_network.spoke_vnet.resource_group_name
  virtual_network_name      = data.azurerm_virtual_network.spoke_vnet.name
  remote_virtual_network_id = azurerm_virtual_network.nonprod_vnet.id
}

```
I always get the error message when running the pipeline in AzureDevOps:

> 
> │ Error: Provider configuration not present
> │ 
> │ To work with azurerm_virtual_network_peering.spoke2nonprod_peering (orphan)
> │ its original provider configuration at
> │ **provider["registry.terraform.io/hashicorp/azurerm"].sharedsubscription** is
> │ required, but it has been removed. This occurs when a provider
> │ configuration is removed while objects created by that provider still exist
> │ in the state. Re-add the provider configuration to destroy
> │ azurerm_virtual_network_peering.spoke2nonprod_peering (orphan), after which
> │ you can remove the provider configuration again.
> ╵
> ##[error]Error: TerraformPlanFailed 1

The same terraform plan works fine if executed locally:

![image](https://user-images.githubusercontent.com/62987024/200678793-acb2797c-4853-4382-9704-34613889b0ff.png)

**Comment by AminTi** [comment]
Thanks for reporting, we'll dig into this.

**Comment by mericstam** [comment]
Is your SPN configured to have access to both subscriptions?

**Comment by marco-svitol** [comment]
Yes, just for testing I am using the alias provider that points to the same subscription as the default provider.

**Comment by jaredfholgate** [comment]
It's not clear what the issue was here, but guessing it is no longer a problem given the age. Closing for now a
